sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -189,15 +189,16 @@ def deserialize_properties(props_struct: struct_pb2.Struct) -> Any:
             if "uri" in props_struct:
                 return known_types.new_remote_asset(props_struct["uri"])
             raise AssertionError("Invalid asset encountered when unmarshaling resource property")
-        elif props_struct[_special_sig_key] == _special_archive_sig:
+        if props_struct[_special_sig_key] == _special_archive_sig:
             # This is an archive. Re-hydrate this object into an Archive.
             if "assets" in props_struct:
                 return known_types.new_asset_archive(deserialize_property(props_struct["assets"]))
             if "path" in props_struct:
                 return known_types.new_file_archive(props_struct["path"])
             if "uri" in props_struct:
                 return known_types.new_remote_archive(props_struct["uri"])
-        elif props_struct[_special_sig_key] == _special_secret_sig:
+            raise AssertionError("Invalid archive encountered when unmarshaling resource property")
+        if props_struct[_special_sig_key] == _special_secret_sig:
```

Comment:
did this change logic, or was this just personal preference?

Reply:
New version of linter made me do it.  

https://github.com/pulumi/pulumi/pull/2900/commits/dcba1e342e17facfdc675b7400e75fe354121289